qtglean/src/cpp/uuid.cpp

```diff
@@ -0,0 +1,39 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "glean/uuid.h"
+
+#include <QDebug>
+
+#if not(defined(__wasm__) || defined(BUILD_QMAKE))
+#  include "qtglean.h"
+#endif
+
+UuidMetric::UuidMetric(int id) : m_id(id) {}
+
+void UuidMetric::set(const QString& uuid) const {
+#if not(defined(__wasm__) || defined(BUILD_QMAKE))
+  return glean_uuid_set(m_id, uuid.toLocal8Bit());
+#endif
+}
+
+QString UuidMetric::generateAndSet() const {
+#if not(defined(__wasm__) || defined(BUILD_QMAKE))
+  return glean_uuid_generate_and_set(m_id);
+#endif
+}
+
+int32_t UuidMetric::testGetNumRecordedErrors(ErrorType errorType) const {
+#if not(defined(__wasm__) || defined(BUILD_QMAKE))
+  return glean_uuid_test_get_num_recorded_errors(m_id, errorType);
+#endif
+  return 0;
+}
+
+QString UuidMetric::testGetValue(const QString& pingName) const {
```

Comment:
You are not using these test methods. Can you land tests together with this PR?

Reply:
These tests are for testing the actual collection of concrete metrics. We should ideally have tests with generic metrics, howeve due to bandwidth and other priorities we did not implement that. It is expected that when new collection is implemented it will have tests and by consequence will test that this API is working. All of this is to say: I would say it is out of scope for this PR, to test this, but we also should leave it here.